pkg/logs/input/container/scanner.go

```diff
@@ -0,0 +1,34 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package container
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/logs/auditor"
+	"github.com/DataDog/datadog-agent/pkg/logs/config"
+	"github.com/DataDog/datadog-agent/pkg/logs/input/docker"
+	"github.com/DataDog/datadog-agent/pkg/logs/input/kubernetes"
+	"github.com/DataDog/datadog-agent/pkg/logs/pipeline"
+	"github.com/DataDog/datadog-agent/pkg/logs/restart"
+)
+
+// NewScanner returns a new container scanner.
+func NewScanner(sources *config.LogSources, pp pipeline.Provider, auditor *auditor.Auditor) restart.Restartable {
```

Comment:
I think this deserves a better explanation.

- when people set up a docker or kubernetes integration, we use it. If none is found but the `container_collect_all` option is set, we try to use the kubernetes integration in k8 environments, and fallback on docker otherwise

Reply:
Still think this